addons/mrp/tests/test_byproduct.py

```diff
@@ -205,3 +206,59 @@ def test_byproduct_putaway(self):
         finished_move_line = mo.move_finished_ids.filtered(lambda m: m.product_id == self.product_a).move_line_ids
         self.assertEqual(byproduct_move_line.location_dest_id, shelf2_location)
         self.assertEqual(finished_move_line.location_dest_id, shelf2_location)
+
+    def test_check_byproducts(self):
+        """
+        Test the byproducts have total cost_share > 100% or < 0% will raise constrains.
+        """
```

Comment:
Your test is good (thank you for writing one :bow:). Just a little English cleanup

```suggestion
    def test_check_byproducts_cost_share(self):
        """
        Test that byproducts with total cost_share > 100% or a cost_share < 0% 
        will throw a ValidationError
        """
 ```